src/hypothesis/_settings.py

```diff
@@ -481,13 +482,42 @@ class Phase(IntEnum):
 
 @unique
 class HealthCheck(Enum):
+    """Classes of health check that Hypothesis performs, for use in.
+
+    :attr:`~hypothesis.settings.suppress_health_check`
+
+    """
+
     exception_in_generation = 0
+    """Deprecated and no longer does anything. It used to convert errors in
+    data generation into FailedHealthCheck error"""
+
     data_too_large = 1
+    """Check for when the typical size of the examples you are generating
+    exceeds the configured maximum size too often. You may wish to raise
+    :attr:`~hypothesis.settings.buffer_size` in preference to supppressing this
+    health check."""
+
     filter_too_much = 2
+    """Check for when the test is filtering out too many examples, either
+    through use of assume() or filter(), or occasionally for Hypothesis
```

Comment:
:func:`~hypothesis.assume`

I'd also like to make `filter()` a hyperlink, which will require a slightly more involved reference to the relevant section in `data.rst`.